{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "3d0beea1-2ed5-4d8b-823e-30627b2d6a06",
   "metadata": {},
   "source": [
    "### We have run the following code in dnanexus platform https://www.dnanexus.com/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "311fdfd5-d43f-4eb0-bf73-3a3d2dcc0f0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Input data needed for running the code which are already available in the platform: \n",
    "#                 1) ukb22828_c1_b0_v3.bgen\n",
    "#                 2) ukb22828_c1_b0_v3.sample\n",
    "#                 3) ukb22418_c1_b0_v2.bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f45d5f26-5b15-43ed-bbd4-7013c94de344",
   "metadata": {},
   "outputs": [],
   "source": [
    "awk '{print $2}' ukb22418_c1_b0_v2.bim | plink2 --bgen ukb22828_c1_b0_v3.bgen ref-first --sample ukb22828_c1_b0_v3.sample --extract /dev/stdin --threads 96 --memory 189000 --make-bed --out ImpQcCh1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "293b14f5-b99f-42e2-98bb-04225af0a87e",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Output after running the above code\n",
    "#           1) ImpQcCh1.log\n",
    "#           2) ImpQcCh1.fam\n",
    "#           3) ImpQcCh1.bim\n",
    "#           4) ImpQcCh1.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a1ca41c5-3eba-4b46-97bc-ca3e2e802e80",
   "metadata": {},
   "outputs": [],
   "source": [
    "## You need to run the above code for 22 times for 22 chromosomes\n",
    "# If you run it for chromosome 8, Changes needed in the above code: change c1 by c8 and Ch1 by Ch8\n",
    "### Similar change for all other chromosomes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5c7d81ff-ab00-46bf-a8c5-4aae0082ec84",
   "metadata": {},
   "source": [
    "### LD pruning using plink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b371183a-d21f-4ef8-ad01-c143f1593a1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "## After running the above code for all chromosomes, run the code below for LD pruning"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "a59249be-a894-4bc1-9004-caf8f0ad620b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh1.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh1\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh1\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "58859 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh1.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998382.\n",
      "58859 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "2%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "73%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "98%\n",
      "99%\n",
      "Pruned 14588 variants from chromosome 1, leaving 44271.\n",
      "Pruning complete.  14588 of 58859 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh1.prune.in and PrunedCh1.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/2\n",
      "\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh1.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh1\n",
      "  --extract PrunedCh1.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh1\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "58859 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh1.nosex .\n",
      "--extract: 44315 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998239.\n",
      "44315 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh1.bed + ImpQcLDpCh1.bim + ImpQcLDpCh1.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 117 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh2.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh2\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh2\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "58295 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh2.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99842.\n",
      "58295 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 15170 variants from chromosome 2, leaving 43125.\n",
      "Pruning complete.  15170 of 58295 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh2.prune.in and PrunedCh2.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 112 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh2.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh2\n",
      "  --extract PrunedCh2.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh2\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "58295 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh2.nosex .\n",
      "--extract: 43169 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998376.\n",
      "43169 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh2.bed + ImpQcLDpCh2.bim + ImpQcLDpCh2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh3.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh3\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh3\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "48783 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh3.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998404.\n",
      "48783 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "15%\n",
      "16%\n",
      "17%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "98%\n",
      "99%\n",
      "Pruned 12437 variants from chromosome 3, leaving 36346.\n",
      "Pruning complete.  12437 of 48783 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh3.prune.in and PrunedCh3.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh3.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh3\n",
      "  --extract PrunedCh3.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh3\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "48783 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh3.nosex .\n",
      "--extract: 36388 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.9983.\n",
      "36388 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh3.bed + ImpQcLDpCh3.bim + ImpQcLDpCh3.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 100 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh4.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh4\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh4\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "45649 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh4.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998303.\n",
      "45649 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "46%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "63%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "73%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "83%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "90%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 12030 variants from chromosome 4, leaving 33619.\n",
      "Pruning complete.  12030 of 45649 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh4.prune.in and PrunedCh4.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh4.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh4\n",
      "  --extract PrunedCh4.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh4\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "45649 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh4.nosex .\n",
      "--extract: 33646 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998224.\n",
      "33646 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh4.bed + ImpQcLDpCh4.bim + ImpQcLDpCh4.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 83 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh5.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh5\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh5\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "43562 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh5.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997892.\n",
      "43562 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "0%\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "56%\n",
      "57%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "64%\n",
      "65%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "72%\n",
      "73%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "80%\n",
      "81%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "88%\n",
      "89%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "96%\n",
      "97%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 11105 variants from chromosome 5, leaving 32457.\n",
      "Pruning complete.  11105 of 43562 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh5.prune.in and PrunedCh5.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/6\n",
      "\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh5.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh5\n",
      "  --extract PrunedCh5.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh5\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "43562 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh5.nosex .\n",
      "--extract: 32485 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997752.\n",
      "32485 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh5.bed + ImpQcLDpCh5.bim + ImpQcLDpCh5.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 80 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh6.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh6\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh6\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "51239 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh6.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998371.\n",
      "51239 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "37%\n",
      "38%\n",
      "39%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "83%\n",
      "84%\n",
      "85%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 17841 variants from chromosome 6, leaving 33398.\n",
      "Pruning complete.  17841 of 51239 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh6.prune.in and PrunedCh6.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 94 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh6.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh6\n",
      "  --extract PrunedCh6.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh6\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "51239 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh6.nosex .\n",
      "--extract: 33444 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998249.\n",
      "33444 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh6.bed + ImpQcLDpCh6.bim + ImpQcLDpCh6.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh7.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh7\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh7\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "39836 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh7.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998311.\n",
      "39836 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "2%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "17%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "63%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "69%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "90%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 10258 variants from chromosome 7, leaving 29578.\n",
      "Pruning complete.  10258 of 39836 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh7.prune.in and PrunedCh7.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 87 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh7.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh7\n",
      "  --extract PrunedCh7.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh7\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "39836 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh7.nosex .\n",
      "--extract: 29600 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99821.\n",
      "29600 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh7.bed + ImpQcLDpCh7.bim + ImpQcLDpCh7.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh8.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh8\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh8\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "37043 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh8.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998392.\n",
      "37043 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "0%\n",
      "1%\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "11%\n",
      "12%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "23%\n",
      "24%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "34%\n",
      "35%\n",
      "35%\n",
      "36%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "46%\n",
      "46%\n",
      "47%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "58%\n",
      "59%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "69%\n",
      "70%\n",
      "70%\n",
      "71%\n",
      "71%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "81%\n",
      "82%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "93%\n",
      "94%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 9530 variants from chromosome 8, leaving 27513.\n",
      "Pruning complete.  9530 of 37043 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh8.prune.in and PrunedCh8.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh8.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh8\n",
      "  --extract PrunedCh8.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh8\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "37043 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh8.nosex .\n",
      "--extract: 27532 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998313.\n",
      "27532 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh8.bed + ImpQcLDpCh8.bim + ImpQcLDpCh8.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 86 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh9.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh9\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh9\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "31833 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh9.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998257.\n",
      "31833 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 7832 variants from chromosome 9, leaving 24001.\n",
      "Pruning complete.  7832 of 31833 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh9.prune.in and PrunedCh9.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/10\n",
      "\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh9.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh9\n",
      "  --extract PrunedCh9.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh9\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "31833 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh9.nosex .\n",
      "--extract: 24027 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998162.\n",
      "24027 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh9.bed + ImpQcLDpCh9.bim + ImpQcLDpCh9.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 89 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh10.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh10\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh10\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "36185 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh10.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998275.\n",
      "36185 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "2%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "21%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "37%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "83%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 9320 variants from chromosome 10, leaving 26865.\n",
      "Pruning complete.  9320 of 36185 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh10.prune.in and PrunedCh10.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh10.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh10\n",
      "  --extract PrunedCh10.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh10\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "36185 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh10.nosex .\n",
      "--extract: 26894 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998193.\n",
      "26894 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh10.bed + ImpQcLDpCh10.bim + ImpQcLDpCh10.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 87 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh11.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh11\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh11\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "36705 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh11.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998455.\n",
      "36705 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "21%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "65%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "92%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 9645 variants from chromosome 11, leaving 27060.\n",
      "Pruning complete.  9645 of 36705 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh11.prune.in and PrunedCh11.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 89 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh11.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh11\n",
      "  --extract PrunedCh11.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh11\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "36705 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh11.nosex .\n",
      "--extract: 27102 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998377.\n",
      "27102 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh11.bed + ImpQcLDpCh11.bim + ImpQcLDpCh11.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh12.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh12\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh12\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "34863 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh12.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998321.\n",
      "34863 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "2%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "17%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "39%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "46%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "90%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 8768 variants from chromosome 12, leaving 26095.\n",
      "Pruning complete.  8768 of 34863 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh12.prune.in and PrunedCh12.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/13\n",
      "\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh12.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh12\n",
      "  --extract PrunedCh12.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh12\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "34863 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh12.nosex .\n",
      "--extract: 26120 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998249.\n",
      "26120 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh12.bed + ImpQcLDpCh12.bim + ImpQcLDpCh12.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 89 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh13.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh13\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh13\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "24463 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh13.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99808.\n",
      "24463 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "10%\n",
      "11%\n",
      "11%\n",
      "12%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "24%\n",
      "25%\n",
      "25%\n",
      "26%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "37%\n",
      "38%\n",
      "38%\n",
      "39%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "51%\n",
      "52%\n",
      "52%\n",
      "53%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "64%\n",
      "65%\n",
      "65%\n",
      "66%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "78%\n",
      "79%\n",
      "79%\n",
      "80%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "91%\n",
      "92%\n",
      "92%\n",
      "93%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 5869 variants from chromosome 13, leaving 18594.\n",
      "Pruning complete.  5869 of 24463 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh13.prune.in and PrunedCh13.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh13.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh13\n",
      "  --extract PrunedCh13.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh13\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "24463 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh13.nosex .\n",
      "--extract: 18614 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998057.\n",
      "18614 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh13.bed + ImpQcLDpCh13.bim + ImpQcLDpCh13.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 56 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh14.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh14\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh14\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "23761 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh14.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997922.\n",
      "23761 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "74%\n",
      "75%\n",
      "76%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 5834 variants from chromosome 14, leaving 17927.\n",
      "Pruning complete.  5834 of 23761 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh14.prune.in and PrunedCh14.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh14.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh14\n",
      "  --extract PrunedCh14.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh14\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "23761 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh14.nosex .\n",
      "--extract: 17947 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997868.\n",
      "17947 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh14.bed + ImpQcLDpCh14.bim + ImpQcLDpCh14.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "15\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 50 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh15.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh15\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh15\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "23099 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh15.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998073.\n",
      "23099 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "0%\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "15%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "65%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "85%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "90%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 5822 variants from chromosome 15, leaving 17277.\n",
      "Pruning complete.  5822 of 23099 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh15.prune.in and PrunedCh15.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh15.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh15\n",
      "  --extract PrunedCh15.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh15\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "23099 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh15.nosex .\n",
      "--extract: 17300 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997986.\n",
      "17300 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh15.bed + ImpQcLDpCh15.bim + ImpQcLDpCh15.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "16\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 63 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh16.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh16\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh16\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "26454 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh16.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998198.\n",
      "26454 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "8%\n",
      "9%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "19%\n",
      "20%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "30%\n",
      "31%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "41%\n",
      "42%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "52%\n",
      "53%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "63%\n",
      "64%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "73%\n",
      "73%\n",
      "74%\n",
      "74%\n",
      "75%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "84%\n",
      "85%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "95%\n",
      "96%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 6449 variants from chromosome 16, leaving 20005.\n",
      "Pruning complete.  6449 of 26454 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh16.prune.in and PrunedCh16.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh16.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh16\n",
      "  --extract PrunedCh16.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh16\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "26454 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh16.nosex .\n",
      "--extract: 20027 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998108.\n",
      "20027 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh16.bed + ImpQcLDpCh16.bim + ImpQcLDpCh16.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "17\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 66 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh17.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh17\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh17\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "24707 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh17.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998161.\n",
      "24707 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "21%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "74%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 6080 variants from chromosome 17, leaving 18627.\n",
      "Pruning complete.  6080 of 24707 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh17.prune.in and PrunedCh17.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh17.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh17\n",
      "  --extract PrunedCh17.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh17\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "24707 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh17.nosex .\n",
      "--extract: 18651 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998084.\n",
      "18651 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh17.bed + ImpQcLDpCh17.bim + ImpQcLDpCh17.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 54 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh18.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh18\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh18\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "21120 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh18.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998126.\n",
      "21120 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "37%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "83%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 4849 variants from chromosome 18, leaving 16271.\n",
      "Pruning complete.  4849 of 21120 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh18.prune.in and PrunedCh18.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/19\n",
      "\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh18.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh18\n",
      "  --extract PrunedCh18.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh18\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "21120 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh18.nosex .\n",
      "--extract: 16282 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998057.\n",
      "16282 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh18.bed + ImpQcLDpCh18.bim + ImpQcLDpCh18.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 44 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh19.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh19\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh19\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "22439 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh19.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99768.\n",
      "22439 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "6%\n",
      "7%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "14%\n",
      "15%\n",
      "15%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "22%\n",
      "23%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "39%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "47%\n",
      "48%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "55%\n",
      "56%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "63%\n",
      "64%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "88%\n",
      "89%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "96%\n",
      "97%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 6139 variants from chromosome 19, leaving 16300.\n",
      "Pruning complete.  6139 of 22439 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh19.prune.in and PrunedCh19.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh19.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh19\n",
      "  --extract PrunedCh19.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh19\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "22439 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh19.nosex .\n",
      "--extract: 16317 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997691.\n",
      "16317 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh19.bed + ImpQcLDpCh19.bim + ImpQcLDpCh19.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 46 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh20.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh20\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh20\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "18885 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh20.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998279.\n",
      "18885 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "46%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "90%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 4380 variants from chromosome 20, leaving 14505.\n",
      "Pruning complete.  4380 of 18885 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh20.prune.in and PrunedCh20.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh20.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh20\n",
      "  --extract PrunedCh20.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh20\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "18885 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh20.nosex .\n",
      "--extract: 14520 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998131.\n",
      "14520 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh20.bed + ImpQcLDpCh20.bim + ImpQcLDpCh20.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 29 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh21.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh21\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh21\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "10768 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh21.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998003.\n",
      "10768 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "21%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "29%\n",
      "30%\n",
      "31%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "44%\n",
      "45%\n",
      "46%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "59%\n",
      "60%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "69%\n",
      "70%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "76%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "83%\n",
      "84%\n",
      "85%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "98%\n",
      "99%\n",
      "Pruned 2553 variants from chromosome 21, leaving 8215.\n",
      "Pruning complete.  2553 of 10768 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh21.prune.in and PrunedCh21.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh21.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh21\n",
      "  --extract PrunedCh21.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh21\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "10768 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh21.nosex .\n",
      "--extract: 8221 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997814.\n",
      "8221 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh21.bed + ImpQcLDpCh21.bim + ImpQcLDpCh21.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 23 duplicate IDs in --extract file.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to PrunedCh22.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh22\n",
      "  --indep-pairwise 50 5 0.5\n",
      "  --out PrunedCh22\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "11936 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to PrunedCh22.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99817.\n",
      "11936 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "\n",
      "1%\n",
      "2%\n",
      "3%\n",
      "4%\n",
      "5%\n",
      "6%\n",
      "6%\n",
      "7%\n",
      "8%\n",
      "9%\n",
      "10%\n",
      "11%\n",
      "12%\n",
      "13%\n",
      "14%\n",
      "14%\n",
      "15%\n",
      "16%\n",
      "17%\n",
      "18%\n",
      "19%\n",
      "20%\n",
      "21%\n",
      "21%\n",
      "22%\n",
      "22%\n",
      "23%\n",
      "24%\n",
      "25%\n",
      "26%\n",
      "27%\n",
      "28%\n",
      "29%\n",
      "29%\n",
      "30%\n",
      "30%\n",
      "31%\n",
      "32%\n",
      "33%\n",
      "34%\n",
      "35%\n",
      "36%\n",
      "37%\n",
      "37%\n",
      "38%\n",
      "38%\n",
      "39%\n",
      "40%\n",
      "41%\n",
      "42%\n",
      "43%\n",
      "44%\n",
      "45%\n",
      "45%\n",
      "46%\n",
      "47%\n",
      "48%\n",
      "49%\n",
      "50%\n",
      "51%\n",
      "52%\n",
      "53%\n",
      "53%\n",
      "54%\n",
      "55%\n",
      "56%\n",
      "57%\n",
      "58%\n",
      "59%\n",
      "60%\n",
      "60%\n",
      "61%\n",
      "61%\n",
      "62%\n",
      "63%\n",
      "64%\n",
      "65%\n",
      "66%\n",
      "67%\n",
      "68%\n",
      "68%\n",
      "69%\n",
      "69%\n",
      "70%\n",
      "71%\n",
      "72%\n",
      "73%\n",
      "74%\n",
      "75%\n",
      "76%\n",
      "76%\n",
      "77%\n",
      "77%\n",
      "78%\n",
      "79%\n",
      "80%\n",
      "81%\n",
      "82%\n",
      "83%\n",
      "84%\n",
      "84%\n",
      "85%\n",
      "86%\n",
      "87%\n",
      "88%\n",
      "89%\n",
      "90%\n",
      "91%\n",
      "92%\n",
      "92%\n",
      "93%\n",
      "94%\n",
      "95%\n",
      "96%\n",
      "97%\n",
      "98%\n",
      "99%\n",
      "Pruned 2882 variants from chromosome 22, leaving 9054.\n",
      "Pruning complete.  2882 of 11936 variants removed.\n",
      "Writing...\n",
      "Marker lists written to PrunedCh22.prune.in and PrunedCh22.prune.out .\n",
      "PLINK v1.90b7.2 64-bit (11 Dec 2023)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ImpQcLDpCh22.log.\n",
      "Options in effect:\n",
      "  --bfile ImpQcCh22\n",
      "  --extract PrunedCh22.prune.in\n",
      "  --make-bed\n",
      "  --out ImpQcLDpCh22\n",
      "\n",
      "32689 MB RAM detected; reserving 16344 MB for main workspace.\n",
      "11936 variants loaded from .bim file.\n",
      "487409 people (222956 males, 264246 females, 207 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ImpQcLDpCh22.nosex .\n",
      "--extract: 9059 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 487409 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99795.\n",
      "9059 variants and 487409 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ImpQcLDpCh22.bed + ImpQcLDpCh22.bim + ImpQcLDpCh22.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning: At least 27 duplicate IDs in --extract file.\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "os.chdir(\"E:\\\\GWAS\\\\plink_win64_20231211\")\n",
    "for chrno in range(1,23):\n",
    "    print(chrno);\n",
    "    filenamein = f'ImpQcCh{chrno}'\n",
    "    filenameout = f'ImpQcLDpCh{chrno}'\n",
    "    filenameint = f'PrunedCh{chrno}'\n",
    "    # Perform LD pruning and generate pruned_data.prune.in\n",
    "    !plink --bfile {filenamein} --indep-pairwise 50 5 0.5 --out {filenameint}\n",
    "    # Step 2: Filter the BED file using the pruned SNPs\n",
    "    !plink --bfile {filenamein} --extract {filenameint}.prune.in --make-bed --out {filenameout}"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
